### PR TITLE
Fix: Update Policy Check script to work on both APFS and CoreStorage volumes

### PR DESCRIPTION
**Pull request by jordanleven**
# Description

This PR fixes an issue where users who are using APFS-formatted drives for Time Machine are unable to verify their encryption status using the `policy-check.sh` script.

Additionally, this PR adds support for failing validation when users have registered backup drives that are not currently connected to their computer (where we're unable to verify encryption status) and fixes an issue where unencrypted mounted Core Storage volumes that are used for backups would cause the Time Machine validation to fail.

# Validation Steps

## Validate Bug
1. If possible, connect two storage volumes to your machine (one APFS and one HFS+).
1. Register both volumes as targets for Time Machine, and do not encrypted backups. A backup does **not** need to complete in order to proceed through validation.
1. Run the current policy check script:
```sh
curl -s https://raw.githubusercontent.com/sparkbox/standard/main/security/security_policy_compliance/policy-checks.sh | sh
```
- [ ] Confirm the validation fails to recognize your APFS volume.

## Validate Fix
1. Run the new policy check script:
```sh
curl -s https://raw.githubusercontent.com/sparkbox/standard/fix--policy-check-apfs/security/security_policy_compliance/policy-checks.sh | sh
```
- [ ] Confirm the validation recognizes all registered volumes and fails validation since both volumes are unencrypted.
2. Reformat both the APFS and HFS drives. Re-register both volumes as targets in Time Machine, but encrypt backups. A backup does **not** need to complete in order to proceed through validation.
1. Run the new policy check script:
```sh
curl -s https://raw.githubusercontent.com/sparkbox/standard/fix--policy-check-apfs/security/security_policy_compliance/policy-checks.sh | sh
```
- [ ] Confirm the validation recognizes all registered volumes and passes validation since both volumes are encrypted.

### CODE REVIEW

File: security/security_policy_compliance/policy-checks.sh [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-  echo "✅";
-  echo "${DETAILS}"
-  echo ""
+  echo "✅ ${DETAILS}\n";
+}
+
+function get_volume_attribute {
+  NEEDLE=$1
+  HAYSTACK=$2
+  echo $(echo "$HAYSTACK" | grep -e "$NEEDLE:" | sed "s/$NEEDLE: *//" | sed 's/^ *//g')
```
Comment (gtempus) [inline security/security_policy_compliance/policy-checks.sh:24]: FWIW, a fancy way to split the line is with `cut`
```suggestion
  echo $(echo "$HAYSTACK" | grep -e "$NEEDLE:" | cut -d : -f2 | sed 's/^ *//g')
```

**Review by gtempus — COMMENTED** [review]
This looks good to me, @jordanleven! I just threw in an option for your needle/haystack pipeline for fun.

File: security/security_policy_compliance/policy-checks.sh (hunk 1 above)
Comment (jordanleven) [reply security/security_policy_compliance/policy-checks.sh]: Awesome - thanks Gary! All updated in 8f4ec438d0c0bbf3d838f1dfe1370060313eca88.

**Review by dustin-jw — APPROVED** [review]
Can confirm that the old script failed on my APFS drive (don't have an HFS one to test), but the new one works as expected. It failed when my drive was unencrypted and succeeded when it was encrypted. Nice work!

**Review by mcksheridan — APPROVED** [review]
I followed the validation process, and the old script fails to recognize my APFS backup but the new script handles it without any issues. Thanks for taking the time out to update this!

**Pull request merged**